api/src/test/kotlin/com/nikitahohulia/listeningplatform/NatsMovieControllerTests.kt

```diff
@@ -0,0 +1,198 @@
+import com.nikitahohulia.nats.NatsSubject.User.GET_ALL
+import com.nikitahohulia.nats.NatsSubject.User.GET_BY_ID
+import com.nikitahohulia.nats.NatsSubject.User.GET_BY_USERNAME
+import com.nikitahohulia.nats.NatsSubject.User.UPDATE
+import com.nikitahohulia.nats.reqreply.user.create.proto.CreateUserRequestCommon
+import com.nikitahohulia.nats.reqreply.user.create.proto.CreateUserResponseCommon
+import com.nikitahohulia.nats.reqreply.user.get_all.proto.GetAllUsersResponseCommon
+import com.nikitahohulia.nats.reqreply.user.get_by_id.proto.DeleteUserByIdRequestCommon
+import com.nikitahohulia.nats.reqreply.user.get_by_id.proto.DeleteUserByIdResponseCommon
+import com.nikitahohulia.nats.reqreply.user.get_by_id.proto.DeleteUserByUsernameRequestCommon
+import com.nikitahohulia.nats.reqreply.user.get_by_id.proto.GetUserByIdRequestCommon
+import com.nikitahohulia.nats.reqreply.user.get_by_id.proto.GetUserByIdResponseCommon
+import com.nikitahohulia.nats.reqreply.user.get_by_id.proto.GetUserByUsernameRequestCommon
+import com.nikitahohulia.nats.reqreply.user.get_by_id.proto.GetUserByUsernameResponseCommon
+import org.bson.types.ObjectId
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Order
+import org.junit.jupiter.api.Test
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNotNull
+import com.google.protobuf.GeneratedMessageV3
+import com.google.protobuf.Parser
+import com.nikitahohulia.nats.commonmodels.user.UserList
+import com.nikitahohulia.nats.reqreply.user.create.proto.UpdateUserRequestCommon
+import com.nikitahohulia.nats.reqreply.user.create.proto.UpdateUserResponseCommon
+import com.nikitahohulia.nats.reqreply.user.get_all.proto.GetAllUsersRequestCommon
+import com.nikitahohulia.nats.reqreply.user.get_by_id.proto.DeleteUserByUsernameResponseCommon
+
+@SpringBootTest
+class NatsMovieControllerTests {
+
+    @Autowired
+    private lateinit var natsConnection: Connection
+
+    @Autowired
+    private lateinit var userRepository: CustomUserRepository
+
+    @AfterEach
+    fun clean() {
+        userRepository.deleteById(ObjectId(userId))
+    }
+
+    private val userId = "651c6a8763d50fb3f7f1ec7b"
+
+    private val userToAdd: com.nikitahohulia.listeningplatform.entity.User =
+        com.nikitahohulia.listeningplatform.entity.User(
+            id = ObjectId(userId),
+            username = "NatsTestUser",
+            password = "123password",
+            email = "nats@gmail.com"
+        )
+
+    @Test
+    fun addUserTestOk() {
+        val proto = userToAdd.toProto()
+        val message = CreateUserRequestCommon.newBuilder().setUser(proto).build()
+
+        val actual = doRequest(CREATE, message, CreateUserResponseCommon.parser())
+        val expected = CreateUserResponseCommon.newBuilder().apply {
+            successBuilder.setUser(proto)
+        }.build()
+
+        assertThat(expected.success.user.toRequest()).isEqualTo(actual.success.user.toRequest())
```

Comment:
what's the purpose of mapping user to request? prepare your expected model in the way, so assertion will look like this
```suggestion
        assertThat(actual).isEqualTo(expected)
```